saleor/order/calculations.py

```diff
@@ -0,0 +1,261 @@
+                continue
+            product = variant.product
+
+            unit_price = manager.calculate_order_line_unit(
+                order, line, variant, product
+            )
+            total_price = manager.calculate_order_line_total(
+                order, line, variant, product
+            )
+            tax_rate = manager.get_order_line_tax_rate(
+                order, product, variant, None, unit_price
+            )
+
+            tax_lines_data.append(
+                TaxLineData(
+                    id=line.id,
+                    currency=currency,
+                    tax_rate=tax_rate,
+                    unit_net_amount=unit_price.net.amount,
+                    unit_gross_amount=unit_price.gross.amount,
+                    total_net_amount=total_price.net.amount,
+                    total_gross_amount=total_price.gross.amount,
+                )
+            )
+            subtotal += total_price
+
+        shipping_price = manager.calculate_order_shipping(order)
+        shipping_tax_rate = manager.get_order_shipping_tax_rate(order, shipping_price)
+        total = shipping_price + subtotal
+
+        return TaxData(
+            currency=currency,
+            total_net_amount=total.net.amount,
+            total_gross_amount=total.gross.amount,
+            subtotal_net_amount=subtotal.net.amount,
+            subtotal_gross_amount=subtotal.gross.amount,
+            shipping_price_net_amount=shipping_price.net.amount,
+            shipping_price_gross_amount=shipping_price.gross.amount,
+            shipping_tax_rate=shipping_tax_rate,
+            lines=tax_lines_data,
+        )
+    except TaxError:
+        return None
+
+
+def _apply_tax_data(
+    order: Order, lines: Iterable[OrderLine], tax_data: TaxData
+) -> None:
+    def qp(net: Decimal, gross: Decimal) -> TaxedMoney:
+        currency = order.currency
+        return quantize_price(
+            TaxedMoney(net=Money(net, currency), gross=Money(gross, currency)),
+            currency,
+        )
+
+    order.total = qp(net=tax_data.total_net_amount, gross=tax_data.total_gross_amount)
+    order.shipping_price = qp(
+        net=tax_data.shipping_price_net_amount,
+        gross=tax_data.shipping_price_gross_amount,
+    )
+    order.shipping_tax_rate = tax_data.shipping_tax_rate
+
+    tax_lines = {line.id: line for line in tax_data.lines}
+    zipped_order_and_tax_lines = ((line, tax_lines[line.id]) for line in lines)
+
+    for (order_line, tax_line) in zipped_order_and_tax_lines:
+        order_line.unit_price = qp(
+            net=tax_line.unit_net_amount, gross=tax_line.unit_gross_amount
+        )
+        order_line.total_price = qp(
+            net=tax_line.total_net_amount, gross=tax_line.total_gross_amount
+        )
+        order_line.tax_rate = tax_line.tax_rate
+
+
+def fetch_order_prices_if_expired(
+    order: Order,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
```

Comment:
I am not conviced that this parameter should be optional. IMHO I would consider treat it as a mandatory parameter, as in many places we already have a prefetched data.